strings/length_longest_substr.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * @file
+ * @brief Length of longest substring (without repeating characters)
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂